packages/@aws-cdk/example-construct-library/lib/example-resource.ts

```diff
@@ -0,0 +1,501 @@
+ *  some examples are shown below.
+ */
+export interface IExampleResource extends
+  // all L2 interfaces need to extend IResource
+  core.IResource,
+
+  // Only for resources that have an associated IAM Role.
+  // Allows this resource to be the target in calls like bucket.grantRead(exampleResource).
+  iam.IGrantable,
+
+  // only for resources that are in a VPC and have SecurityGroups controlling their traffic
+  ec2.IConnectable {
+
+  /**
+   * The ARN of example resource.
+   * Equivalent to doing `{ 'Fn::GetAtt': ['LogicalId', 'Arn' ]}`
+   * in CloudFormation.
+   * It needs to be annotated with '@attribute'.
+   *
+   * @attribute
+   */
+  readonly exampleResourceArn: string;
+
+  /**
+   * The physical name of the example resource.
+   * Often, equivalent to doing `{ 'Ref': 'LogicalId' }`
+   * (but not always - depends on the particular resource modeled)
+   * in CloudFormation.
+   * Also needs to be annotated with '@attribute'.
+   *
+   * @attribute
+   */
+  readonly exampleResourceName: string;
+
+  /**
+   * For resources that have an associated IAM Role,
+   * surface that Role as a property,
+   * so that other classes can add permissions to it.
+   * Make it optional,
+   * as resources imported with {@link ExampleResource.fromExampleResourceName}
+   * will not have this set.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * For resources that have an associated IAM Role,
+   * surface a method that allows you to conditionally
+   * add a statement to that Role if it's known.
+   * This is just a convenience,
+   * so that clients of your interface don't have to check {@link role} for null.
+   * Many such methods in the CDK return void;
+   * you can also return a boolean indicating whether the permissions were in fact added
+   * (so, when {@link role} is not null).
+   */
+  addToRolePolicy(policyStatement: iam.PolicyStatement): boolean;
+
+  /**
+   * An example of a method that grants the given IAM identity
+   * permissions to this resource
+   * (in this case - read permissions).
+   */
+  grantRead(identity: iam.IGrantable): iam.Grant;
+
+  /**
+   * Add a CloudWatch rule that will use this resource as the source of events.
+   * Resources that emit events have a bunch of methods like these,
+   * that allow different resources to be triggered on various events happening to this resource
+   * (like item added, item updated, item deleted, ect.) -
+   * exactly which methods you need depends on the resource you're modeling.
+   */
+  onEvent(id: string, options?: events.OnEventOptions): events.Rule;
+
+  /**
+   * Standard method that allows you to capture metrics emitted by this resource,
+   * and use them in dashboards and alarms.
+   * The details of which metric methods you should have of course depends on the
+   * resource that is being modeled.
+   */
+  metricSth(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric;
```

Comment:
Maybe use an example metric -

```suggestion
  metricCount(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric;
```

Reply:
Sure, done.